### ISSUE TITLE
Issue #1

### ISSUE THREAD

**Issue by simonkivinen** [simonkivinen/AARV--Autonomous_AI_Recognition_Vision#1]
**Labels:** `refactor` `good first issue`
 
Just nu är all kod skriven i ett enda skript uppifrån och ner. Det gör det svårt att återanvända nätverket för olika problem.
 
**Uppgift:**
Paketera nätverket i en Python-klass med följande metoder:
- `__init__(self, layer_sizes)` – tar en lista med antal noder per lager, t.ex. `[2, 4, 1]`
- `forward(self, X)` – framåtpassering
- `backward(self, X, y)` – bakåtpassering och viktuppdatering
- `train(self, X, y, epochs, learning_rate)` – träningsloop
- `predict(self, X)` – gör en gissning med tränade vikter
**Varför:**
När nätverket är en klass kan du skapa ett nytt nätverk för vilket problem som helst med en enda rad:
```python
net = NeuralNetwork([784, 128, 10])  # sifferigenkänning
net = NeuralNetwork([2, 4, 1])       # XOR
```
